pkgs/development/compilers/llvm/common/libcxx/default.nix

```diff
@@ -123,9 +129,28 @@ stdenv.mkDerivation (rec {
   # libc++.so is a linker script which expands to multiple libraries,
   # libc++.so.1 and libc++abi.so or the external cxxabi. ld-wrapper doesn't
   # support linker scripts so the external cxxabi needs to be symlinked in
-  postInstall = lib.optionalString (cxxabi != null) ''
+  postInstall = ''
+    libcxxabi=$out/lib/libc++abi.a
+  '' + lib.optionalString (cxxabi != null) ''
     lndir ${lib.getDev cxxabi}/include $dev/include/c++/v1
     lndir ${lib.getLib cxxabi}/lib $out/lib
+    libcxxabi=$out/lib/lib${cxxabi.libName}.a
+  '' + lib.optionalString (!lib.versionAtLeast release_version "16") ''
```

Comment:
Why doesn't this also check for `cxxabi == null`?

Reply:
`cxxabi != null` is true for freebsd. `nix-build -A pkgsCross.x86_64-freebsd.llvmPackages_17.libcxx` tho need to modify libcxxrt to something like PR https://github.com/NixOS/nixpkgs/pull/306238 so libcxxrt.a exists in the non pkgsStatic case (no idea if freebsd works under pkgsStatic).

however, it seems like that is causing an infinite recursion so some change was committed between now and  https://github.com/NixOS/nixpkgs/pull/305876 which broke freebsd eval.

to drop into a freebsd builder to try linking / building hello.cxx by hand do something like.

`nix-shell -I nixpkgs=. --expr 'with import <nixpkgs> {}; pkgsCross.x86_64-freebsd.stdenv.mkDerivation { name = "abc"; }'`

AFAIK even if you get freebsd to build the libc it generates segfaults when used on actual freebsd, but maybe that will be fixed soon.

Reply:
i agree that this needs the `cxxabi != null` check. if you're going to leave it out for some additional PR the hackery with `libcxxabi=xxx` can be removed and just `$out/lib/libc++abi.a` used instead.